### PR TITLE
add in-code comments

### PR DESCRIPTION
**Pull request by justprosh**

### CODE REVIEW

File: src/impls.rs [hunk 1]
```diff
@@ -59,6 +59,8 @@ fn check_key_existence(connection: &Connection, key: String, current_timestamp_s
     get_key_metadata_helper(&connection, key, current_timestamp_sec).map(|_| ())
 }
 
+
+/// timestamps should be passed from host ("peer" "timestamp_sec")
```
Comment (folex) [inline src/impls.rs:63]:
```suggestion
/// Check timestamps are generated on the current host with builtin ("peer" "timestamp_sec")
```

File: src/impls.rs [hunk 2]
```diff
@@ -68,6 +70,7 @@ pub(crate) fn check_timestamp_tetraplets(call_parameters: &CallParameters, arg_n
         && tetraplet.peer_pk == call_parameters.host_id).then(|| ()).wrap_err(error_msg)
 }
 
+
```
Comment (folex) [inline src/impls.rs:73]: ?

File: src/impls.rs [hunk 3]
```diff
@@ -195,6 +201,7 @@ pub fn register_key_impl(key: String, current_timestamp_sec: u64, pin: bool, wei
     update_key(&get_connection()?, key, peer_id, current_timestamp_sec.clone(), current_timestamp_sec, pin, weight)
 }
 
+/// used for replication, same as register_key, but key.pinned is ignored
```
Comment (folex) [inline src/impls.rs:204]: is it the same as `register_key`? Shouldn't it be different in that it avoids the `peer_id` check?

I'm not sure this check is OK for republishing. Wouldn't it prevent neighborhood from updating `timestamp_accessed`?
```rust
Ok(key) => key.peer_id == peer_id && key.timestamp_created < timestamp_created,
```

Also, I see it sets `timestamp_created` to `current_timestamp_sec`. Seems like a bug.

File: src/impls.rs [hunk 4]
```diff
@@ -204,6 +211,7 @@ pub fn republish_key_impl(key: Key, current_timestamp_sec: u64) -> SqliteResult<
     update_key(&get_connection()?, key.key, key.peer_id, key.timestamp_created, current_timestamp_sec, false, key.weight)
 }
 
+/// put value with caller peer_id if key exists. If value is NOT host and key already has VALUES_LIMIT records value with the smallest weight displaced
```
Comment (folex) [inline src/impls.rs:214]:
```suggestion
/// Put value with caller peer_id if the key exists. 
/// If the value is NOT a host value and the key already has `VALUES_LIMIT` records, then a value with the smallest weight is removed and the new value is inserted instead.
```

File: src/impls.rs [hunk 5]
```diff
@@ -248,6 +256,7 @@ pub fn put_value_impl(key: String, value: String, current_timestamp_sec: u64, re
     }
 }
 
+/// return all values by corresponding key
```
Comment (folex) [inline src/impls.rs:259]:
```suggestion
/// Return all values by key
```

File: src/impls.rs [hunk 6]
```diff
@@ -262,6 +271,7 @@ pub fn get_values_helper(connection: &Connection, key: String) -> SqliteResult<V
     Ok(result)
 }
 
+/// return all values by corresponding key and update timestamp_accessed
```
Comment (folex) [inline src/impls.rs:274]:
```suggestion
/// Return all values by key and update timestamp_accessed
```

File: src/impls.rs [hunk 7]
```diff
@@ -277,6 +287,7 @@ pub fn get_values_impl(key: String, current_timestamp_sec: u64) -> SqliteResult<
     get_values_helper(&connection, key)
 }
 
+/// if key exists merge records with existing (last-write-wins) and put
```
Comment (folex) [inline src/impls.rs:290]:
```suggestion
/// If the key exists, then merge new records with existing (last-write-wins) and put
```

File: src/impls.rs [hunk 8]
```diff
@@ -303,6 +314,7 @@ pub fn republish_values_impl(key: String, mut records: Vec<Record>, current_time
     Ok(updated)
 }
 
+/// remove expired by timestamp_created values and keys. If key is expired but has actual host values it has same lifetime as its host values. Return the number of keys and values removed
```
Comment (folex) [inline src/impls.rs:317]:
```suggestion
/// Remove expired values and expired empty keys. 
/// Expired means  that `timestamp_created` has surpassed `expired_timeout`.
/// Return number of keys and values removed
```

File: src/impls.rs [hunk 9]
```diff
@@ -338,6 +350,7 @@ pub fn clear_expired_impl(current_timestamp_sec: u64) -> SqliteResult<(u64, u64)
     Ok((deleted_keys, deleted_values))
 }
 
+/// return stale by timestamp_accessed keys and values, delete NOT pinned keys and NOT host values
```
Comment (folex) [inline src/impls.rs:353]:
```suggestion
/// Delete all stale keys and values except for pinned keys and host values. 
/// Stale means that `timestamp_accessed` has surpassed `stale_timeout`.
/// Returns all deleted items
```

File: src/impls.rs [hunk 10]
```diff
@@ -426,6 +442,7 @@ pub fn clear_host_value_impl(key: String, current_timestamp_sec: u64) -> SqliteR
     (connection.changes() == 1).as_result((), SqliteError { code: None, message: Some("host value not found".to_string()) })
 }
 
+/// used for replication of host values
```
Comment (folex) [inline src/impls.rs:445]: How?

**Review by folex — CHANGES_REQUESTED** [review]

File: src/impls.rs (hunk 3 above)
Comment (justprosh) [reply src/impls.rs:204]: Nope, it sets `timestamp_accessed` to `current_timestamp_sec` and other constraints checked in `update_key`

Comment (justprosh) [reply src/impls.rs:204]: oh, i see your point about `timestamp_created`

Comment (justprosh) [reply src/impls.rs:204]: In the case of republishing always should "win" the earliest key, I think. But we should discuss this on scrum tomorrow

Comment (folex) [reply src/impls.rs]: Yeah, makes sense.

**Review by folex — APPROVED** [review]

**Pull request merged**